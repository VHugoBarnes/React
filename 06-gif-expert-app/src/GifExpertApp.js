// LecciÃ³n 69. GifExpertApp component
import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    // const categories = ['The Office', 'Minecraft', 'Clash of Clans'];
    const [categories, setCategories] = useState(['The Office', 'Minecraft', 'Clash of Clans']);

    // const handleAdd = () => {
    //     // setCategories( ['Little Witch Academia', ...categories] );
    //     setCategories( cats => ['Little Witch Academia', ...cats] );
    // };

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr></hr>

            <ol>
                {
                    categories.map( category =>{
                    return <li key={category}>{category}</li>
                    } )
                }
            </ol>
        </>
    );

};
