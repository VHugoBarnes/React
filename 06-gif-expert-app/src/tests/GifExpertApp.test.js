import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en el archivo GifExpertApp', () => {
   
    test('Debe renderizar adecuadamente la app con matchSnapshot', () => {
        const wrapper = shallow( <GifExpertApp /> );

       expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe de mostrar una lista de categorias', () => {
        
        const categories = ['Ellie Kemper', 'Minecraft'];

        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )
    });
    
});
