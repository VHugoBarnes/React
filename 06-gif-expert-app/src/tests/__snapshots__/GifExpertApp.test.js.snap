// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Ellie Kemper"
      key="Ellie Kemper"
    />
    <GifGrid
      category="Minecraft"
      key="Minecraft"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el archivo GifExpertApp Debe renderizar adecuadamente la app con matchSnapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
